src/screens/Referral/ConfirmClaimScreen.js

```diff
@@ -0,0 +1,140 @@
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { View } from 'react-native';
+import type { NavigationScreenProp } from 'react-navigation';
+import { BoldText, LightText } from 'components/Typography';
+import { Container } from 'components/Layout';
+import Button from 'components/Button';
+import Header from 'components/Header';
+import styled from 'styled-components/native';
+import AssetPattern from 'components/AssetPattern';
+import { baseColors, fontSizes } from 'utils/variables';
+import { claimTokensAction } from 'actions/referralsActions';
+import type { ClaimTokenAction } from 'actions/referralsActions';
+import ProcessingClaim from './ProcessingClaim';
+import ErrorClaim from './ErrorClaim';
+
+const Center = styled.View`
+  alignItems: stretch;
+  margin-top: 30px
+  margin-bottom: 50px
+  margin-horizontal: 20px;
+`;
+
+const MainWrapper = styled.View`
+  flex: 1;
+  flexDirection: column;
+  justifyContent: space-between;
+`;
+
+const assetData = {
+  token: 'PLR',
+  iconColor: 'https://api-qa-core.pillarproject.io/asset/images/tokens/icons/plrColor.png?size=3',
+};
+
+const TokenValue = styled(BoldText)`
+  font-size: ${fontSizes.giant}px;
+  text-align: center;
+`;
+
+const TextCode = styled(LightText)`
+  font-size: ${fontSizes.medium}px;
+  color: ${baseColors.coolGrey};
+  text-align: center;
+  margin-bottom: 30px
+`;
+
+type Props = {
+  claimTokens: Function,
+  user: Object,
+  navigation: NavigationScreenProp<*>,
+}
+
+type State = {
+  isFetching: boolean,
+  isError: boolean,
+}
+
+
+class ConfirmClaimScreen extends React.Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      isFetching: false,
+      isError: false,
+    };
+  }
+
+  handleClaim = () => {
+    const { navigation, claimTokens, user } = this.props;
+    const code = navigation.getParam('code') ? navigation.getParam('code') : 'No code';
+    this.setState({ isError: false, isFetching: true });
+    claimTokens({ walletId: user.walletId, code }, ({ error }) => {
+      if (error) {
+        this.setState({ isError: true, isFetching: false });
+      } else {
+        navigation.goBack();
+      }
+    });
+  };
+
+  render() {
+    const { navigation } = this.props;
+    const code = navigation.getParam('code') ? navigation.getParam('code') : 'No code';
+    const { isFetching, isError } = this.state;
+    return (
+      <Container>
+        <Header gray title="claim tokens" onBack={() => navigation.goBack(null)} />
+        <MainWrapper>
+          <TextCode>From {code}</TextCode>
+          { isError ? <ErrorClaim /> : null }
```

Comment:
You can simplify this and following ones by changing it like this:
`{isError && <ErrorClaim />}`
(if variable is boolean. If not, it's better to turn it into one (by adding !!), otherwise Android will throw errors of text not being placed inside of `<Text> `component.

Reply:
👍  updated for all 3